File: TP03/listaSequencial.java
import java.util.*;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.text.ParseException;
import java.text.SimpleDateFormat;

class Pokemon {
    private int id;
    private int generation;
    private String name;
    private String description;
    private String[] types;
    private String[] abilities;
    private Double weight;
    private Double height;
    private int captureRate;
    private Boolean isLegendary;
    private String captureDate;

    //construtores
    public Pokemon(int id, int generation, String name, String description, String[] types, String[] abilities,
     Double weight, Double height, int captureRate, Boolean isLegendary, String captureDate) {
        this.id = id;
        this.generation = generation;
        this.name = name;
        this.description = description;
        this.types = types;
        this.abilities = abilities;
        this.weight = weight;
        this.height = height;
        this.captureRate = captureRate;
        this.isLegendary = isLegendary;
        this.captureDate = captureDate;
    }

    public Pokemon() {
        this.id = 0;
        this.generation = 0;
        this.name = "";
        this.description = "";
        this.types = new String[0];
        this.abilities = new String[0];
        this.weight = 0.0;
        this.height = 0.0;
        this.captureRate = 0;
        this.isLegendary = false;
        this.captureDate = "";
    }

    //gets
    public int getId() {
        return id;
    }
    public int getGeneration() {
        return generation;
    }
    public String getName() {
        return name;
    }
    public String getDescription() {
        return description;
    }
    public String[] getTypes() {
        return types;
    }
    public String[] getAbilities() {
        return abilities;
    }
    public Double getWeight() {
        return weight;
    }
    public Double getHeight() {
        return height;
    }
    public int getCaptureRate() {
        return captureRate;
    }
    public Boolean getIsLengendary() {
        return isLegendary;
    }
    public String getCaptureDate() {
        return captureDate;
    }

    //sets
    public void setId(int id) {
        this.id = id;
    }
    public void setGeneration(int generation) {
        this.generation = generation;
    }
    public void setName(String name) {
        this.name = name;
    }
    public void setDescription(String description) {
        this.description = description;
    }
    public void setTypes(String[] types) {
        this.types = types;
    }
    public void setAbilities(String[] abilities) {
        this.abilities = abilities;
    }
    public void setWeight(Double weight) {
        this.weight = weight;
    }
    public void setHeight(Double height) {
        this.height = height;
    }
    public void setCaptureRate(int captureRate) {
        this.captureRate = captureRate;
    }
    public void setIsLegendary(Boolean isLegendary) {
        this.isLegendary = isLegendary;
    }
    public void setCaptureDate(String capturDate) {
        this.captureDate = capturDate;
    }

    //clone
    public Pokemon clone() {
        return new Pokemon(this.id, this.generation, this.name, this.description, 
                           this.types.clone(), this.abilities.clone(), 
                           this.weight, this.height, this.captureRate, 
                           this.isLegendary, this.captureDate);
    }

    //imprimir
    public void imprimir(Pokemon pokemon) {
        System.out.print("[#" + pokemon.getId() + " -> " + pokemon.getName() + ": " + pokemon.getDescription() + " - ");
    
        // Imprimir tipos
        String[] tipos = pokemon.getTypes();
        System.out.print(Arrays.toString(tipos) + " - ");
    
        // Imprimir habilidades
        String[] habilidades = pokemon.getAbilities();
        System.out.print(Arrays.toString(habilidades) + " - ");
    
        // Imprimir outros atributos
        System.out.print(pokemon.getWeight() + "kg - " + pokemon.getHeight() + "m - " +
                         pokemon.getCaptureRate() + "% - " + pokemon.getIsLengendary() + 
                         " - " + pokemon.getGeneration() + " gen] - " + pokemon.getCaptureDate());
        System.out.println();
    }

    //ler
    public void ler() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("ID do pokemon: ");
        this.id = scanner.nextInt();
        System.out.println("Geração do pokemon: ");
        this.generation = scanner.nextInt();
        scanner.nextLine();
        System.out.println("Nome do pokemon: ");
        this.name = scanner.nextLine();
        System.out.println("Descrição do pokemon: ");
        this.description = scanner.nextLine();
        System.out.println("Tipo(s) do pokemon: ");
        this.types = scanner.nextLine().split(",");
        System.out.println("Abilidades do pokemon: ");
        this.abilities = scanner.nextLine().split(",");
        System.out.println("Peso do pokemon: ");
        this.weight = scanner.nextDouble();
        System.out.println("Tamanho do pokemon: ");
        this.height = scanner.nextDouble();
        System.out.println("Chance de captura: ");
        this.captureRate = scanner.nextInt();
        System.out.println("É um pokemon Lendario? ");
        this.isLegendary = scanner.nextBoolean();
        System.out.println("Data de captura: ");
        this.captureDate = scanner.nextLine();
        scanner.close();
    }
}

class Celula {
    public Pokemon pokemon;
    public Celula prox;

    public Celula(Pokemon pokemon) {
        this.pokemon = pokemon;
        this.prox = null;
    }
}

class listaPokemon {
    private Celula primeiro;
    private Celula ultimo;

    public listaPokemon() {
        primeiro = new Celula(null);
        ultimo = primeiro;
    }

    public void inserirInicio(Pokemon pokemon) {
        Celula nova = new Celula(pokemon);
        nova.prox = primeiro.prox;
        primeiro.prox = nova;
        if (primeiro == ultimo) {
            ultimo = nova;
        }
    }

    public void inserir(Pokemon pokemon, int pos) throws Exception {
        int tamanho = tamanho();
        if (pos < 0 || pos > tamanho) {
            throw new Exception("Erro ao inserir posição (" + pos + " / tamanho = " + tamanho + ") inválida!");
        } else if (pos == 0) {
            inserirInicio(pokemon);
        } else if (pos == tamanho) {
            inserirFim(pokemon);
        } else {
            Celula i = primeiro;
            for (int j = 0; j < pos; j++, i = i.prox);
            Celula nova = new Celula(pokemon);   
            nova.prox = i.prox;
            i.prox = nova;
        }
    }

    public void inserirFim(Pokemon pokemon) {
        Celula nova = new Celula(pokemon);
        ultimo.prox = nova;
        ultimo = nova;
    }

    public Pokemon removerInicio() throws Exception {
        if (primeiro == ultimo) {
            throw new Exception("Erro ao remover (vazia)!");
        }
        Celula tmp = primeiro.prox;
        primeiro.prox = tmp.prox;
        if (tmp == ultimo) {
            ultimo = primeiro;
        }
        tmp.prox = null;
        return tmp.pokemon;
    }

    public Pokemon remover(int pos) throws Exception {
        int tamanho = tamanho();
        if (primeiro == ultimo) {
            throw new Exception("Erro ao remover (vazia)!");
        } else if (pos < 0 || pos >= tamanho) {
            throw new Exception("Erro ao remover (posicao " + pos + " / " + tamanho + " invalida!");
        } else if (pos == 0) {
            return removerInicio();
        } else {
            Celula i = primeiro;
            for (int j = 0; j < pos; j++, i = i.prox);
            Celula tmp = i.prox;
            i.prox = tmp.prox;
            if (tmp == ultimo) {
                ultimo = i;
            }
            tmp.prox = null;
            return tmp.pokemon;
        }
    }

    public Pokemon removerFim() throws Exception {
        if (primeiro == ultimo) {
            throw new Exception("Erro ao remover (vazia)!");
        }
        Celula i;
        for (i = primeiro; i.prox != ultimo; i = i.prox);
        Celula tmp = ultimo;
        ultimo = i;
        i.prox = null;
        return tmp.pokemon;
    }

    public void mostrar() {
        int pos = 0;
        for(Celula i = primeiro.prox; i != null; i = i.prox) {
            System.out.print("[" + pos + "] ");
            i.pokemon.imprimir(i.pokemon);
            pos++;
        }
    }

    public int tamanho() {
        int tamanho = 0;
        for (Celula i = primeiro; i.prox != null; i = i.prox, tamanho++);
        return tamanho;
    }
}

public class listaSequencial {
    public static Pokemon parsePokemon(String[] parts) {
        try {
            int id = Integer.parseInt(parts[0].trim());
            int gen = Integer.parseInt(parts[1].trim());
            String name = parts[2].trim();
            String description = parts[3].trim();
            String type1 = parts[4].trim();
            String type2 = parts[5].trim();
            String[] types = type2.isEmpty() ? new String[]{type1} : new String[]{type1, type2};
            String habilidadesString = parts[6].replace(".", ",").replace("[", "").replace("]", "").trim();
            String[] habilidades = habilidadesString.split(",");  
            Double peso = parts[7].trim().equals("0") ? 0.0 : Double.parseDouble(parts[7].trim());
            Double altura = parts[8].trim().equals("0") ? 0.0 : Double.parseDouble(parts[8].trim());
            int captura = Integer.parseInt(parts[9].trim());
            Boolean lendario = parts[10].trim().equals("1");
            String data = parts[11].trim();
    
            return new Pokemon(id, gen, name, description, types, habilidades, peso, altura, captura, lendario, data);
        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
            System.err.println("Erro ao processar o Pokémon com dados: " + Arrays.toString(parts));
            e.printStackTrace();
            return null;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<String> requestedIds = new ArrayList<>();

        listaPokemon lista = new listaPokemon();

        while (true) {
            String input = scanner.nextLine();

            if(input.equals("FIM")) break;

            requestedIds.add(input);
        }

        try{
            BufferedReader reader = new BufferedReader(new FileReader("pokemon.csv"));

            String line = reader.readLine();

            while ((line = reader.readLine()) != null) {
                line = line.replace("\",,,", "\", , ,");
                line = line.replace(",,", ", ,");
                line = line.replace("\"", "");
                line = line.replace(", ", ".");
                line = line.replace(",", ";");
                line = line.replace(".;[", "[; ;");
                line = line.replace("]..;", "];0.0;0.0;");
                line = line.replace("].; ;", "];0.0;0.0;");

                String[] parts = line.split(";");

                Pokemon novoPokemon = parsePokemon(parts);
                if(requestedIds.contains(String.valueOf(novoPokemon.getId()))) {
                    lista.inserirFim(novoPokemon);
                }
            }
            reader.close();
        } catch(IOException e) {
            System.err.println("Erro ao ler o arquivo CSV: " + e.getMessage());
        }

        int n = scanner.nextInt();
        scanner.nextLine();

        for(int i = 0; i < n; i++) {
            String command = scanner.next();

            if(command.equals("II")) {
                String idProcurado = scanner.next();
                scanner.nextLine();
                boolean encontrado = false;

                try {
                    BufferedReader reader = new BufferedReader(new FileReader(("pokemon.csv")));
                    
                    String line;

                    while((line = reader.readLine()) != null) {
                        line = line.replace("\",,,", "\", , ,");
                        line = line.replace(",,", ", ,");
                        line = line.replace("\"", "");
                        line = line.replace(", ", ".");
                        line = line.replace(",", ";");
                        line = line.replace(".;[", "[; ;");
                        line = line.replace("]..;", "];0.0;0.0;");
                        line = line.replace("].; ;", "];0.0;0.0;");

                        String[] parts = line.split(";");

                        String idLido = parts[0];

                        if(idLido.equals(idProcurado)) {
                            Pokemon novoPokemon = parsePokemon(parts);
                            lista.inserirInicio(novoPokemon);
                            encontrado = true;
                            break;
                        }
                    }
                    reader.close();

                    if (!encontrado) {
                        System.err.println("Pokemon com o ID '" + idProcurado + "' não encontrado.");
                    }
                } catch (IOException e) {
                    System.err.println("Erro ao ler o arquivo CSV: " + e.getMessage());
                }
            } else if(command.equals("IF")) {
                String idProcurado = scanner.next();
                scanner.nextLine();
                boolean encontrado = false;

                try {
                    BufferedReader reader = new BufferedReader(new FileReader(("pokemon.csv")));
                    
                    String line;

                    while((line = reader.readLine()) != null) {
                        line = line.replace("\",,,", "\", , ,");
                        line = line.replace(",,", ", ,");
                        line = line.replace("\"", "");
                        line = line.replace(", ", ".");
                        line = line.replace(",", ";");
                        line = line.replace(".;[", "[; ;");
                        line = line.replace("]..;", "];0.0;0.0;");
                        line = line.replace("].; ;", "];0.0;0.0;");

                        String[] parts = line.split(";");

                        String idLido = parts[0];

                        if(idLido.equals(idProcurado)) {
                            Pokemon novoPokemon = parsePokemon(parts);
                            lista.inserirFim(novoPokemon);
                            encontrado = true;
                            break;
                        }
                    }
                    reader.close();

                    if (!encontrado) {
                        System.err.println("Pokemon com o ID '" + idProcurado + "' não encontrado.");
                    }
                } catch (IOException e) {
                    System.err.println("Erro ao ler o arquivo CSV: " + e.getMessage());
                }
            } else if(command.equals("I*")) {
                int pos = scanner.nextInt();
                String idProcurado = scanner.next();
                scanner.nextLine();
                boolean encontrado = false;

                try {
                    BufferedReader reader = new BufferedReader(new FileReader(("pokemon.csv")));
                    
                    String line;

                    while((line = reader.readLine()) != null) {
                        line = line.replace("\",,,", "\", , ,");
                        line = line.replace(",,", ", ,");
                        line = line.replace("\"", "");
                        line = line.replace(", ", ".");
                        line = line.replace(",", ";");
                        line = line.replace(".;[", "[; ;");
                        line = line.replace("]..;", "];0.0;0.0;");
                        line = line.replace("].; ;", "];0.0;0.0;");

                        String[] parts = line.split(";");

                        String idLido = parts[0];

                        if(idLido.equals(idProcurado)) {
                            Pokemon novoPokemon = parsePokemon(parts);
                            try {
                                lista.inserir(novoPokemon, pos);
                            } catch (Exception e) {
                                System.err.println("Erro ao inserir Pokemon na posição " + pos + ": " + e.getMessage());
                            }
                            encontrado = true;
                            break;
                        }
                    }
                    reader.close();

                    if (!encontrado) {
                        System.err.println("Pokemon com o ID '" + idProcurado + "' não encontrado.");
                    }
                } catch (IOException e) {
                    System.err.println("Erro ao ler o arquivo CSV: " + e.getMessage());
                }
            } else if (command.equals("RI")) {
                try {
                    Pokemon removido = lista.removerInicio();
                    System.out.println("(R)" + removido.getName());
                } catch (Exception e) {
                    System.err.println("Erro ao remover do início: " + e.getMessage());
                }
            } else if (command.equals("R*")) {
                int pos = scanner.nextInt();
                scanner.nextLine(); // Para consumir a nova linha
                try {
                    Pokemon removido = lista.remover(pos);
                    System.out.println("(R)" + removido.getName());
                } catch (Exception e) {
                    System.err.println("Erro ao remover na posição " + pos + ": " + e.getMessage());
                }
            } else if (command.equals("RF")) {
                try {
                    Pokemon removido = lista.removerFim();
                    System.out.println("(R)" + removido.getName());
                } catch (Exception e) {
                    System.err.println("Erro ao remover do fim: " + e.getMessage());
                }
            }
            
        }

        scanner.close();
        lista.mostrar();
    }
}

File: TP03/listaSequencial.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <stdlib.h>
#include <time.h>

char ids[801][5];
int numIds = 0;
int numNomes = 0;

typedef struct {
    char id[8];
    char generation[5];
    char name[50];
    char description[100];
    char types[2][30];
    char abilities[1000];
    char weight[10];
    char height[10];
    char captureRate[10];
    bool isLegendary;   
    char captureDate[15];
} Pokemon;

typedef struct {
    Pokemon pokemon;
     struct Celula* prox; 
} Celula;

typedef struct {
    Celula* primeiro;
    Celula* ultimo;
    int tam;
} ListaSequencial;


void start(ListaSequencial* lista) {
    lista -> primeiro = NULL;
    lista -> ultimo = NULL;
    lista -> tam = 0;
}

Celula* novaCelula(Pokemon pokemon) {
    Celula* nova = (Celula*)malloc(sizeof(Celula));
    if(nova == NULL) {
        printf("Falha na alocacao da celula");
        exit(1);
    }
    nova -> pokemon = pokemon;
    nova -> prox = NULL;
    return nova;
}

void inserirInicio(ListaSequencial* lista, Pokemon pokemon) {
    Celula* nova = novaCelula(pokemon);
    nova->prox = lista->primeiro;
    lista->primeiro = nova;
    if(lista->ultimo == NULL) {
        lista->ultimo = nova;
    }
    lista->tam++;
}

void inserirFim(ListaSequencial* lista, Pokemon pokemon) {
    Celula* nova = novaCelula(pokemon);
    if(lista->tam != 0) {
        lista->ultimo->prox = nova;
    }
    lista->ultimo = nova;
    if(lista->primeiro == NULL) {
        lista->primeiro = nova;
    }
    lista->tam++;
}

void inserir(ListaSequencial* lista, Pokemon pokemon, int pos) {
    if(pos > lista->tam) {
        printf("ERRO");
        exit(1);
    }
    Celula* j;
    int i = 0;
    for(j = lista->primeiro, i = 0; i < pos - 1; i++, j = j->prox);

    Celula* tmp = j->prox;
    Celula* nova = novaCelula(pokemon);
    j->prox = nova;
    nova->prox = tmp;
    lista->tam++;
}


void removerInicio(ListaSequencial* lista) {
    if(lista->primeiro == NULL) {
        printf("Erro, Lista vazia");
        exit(1);
    }

    Celula* removida = lista->primeiro;
    printf("(R) %s\n", removida->pokemon.name);
    free(removida);
    lista->primeiro = lista->primeiro->prox;
    lista->tam--;
}

void removerFim(ListaSequencial* lista) {
    if (lista->primeiro == NULL) {
        printf("Erro, Lista vazia");
        exit(1);
    }
    if (lista->primeiro == lista->ultimo) {
        printf("(R) %s\n", lista->primeiro->pokemon.name);
        free(lista->primeiro);
        lista->primeiro = lista->ultimo = NULL;
    } else {
        Celula* i;
        for (i = lista->primeiro; i->prox != lista->ultimo; i = i->prox);
        printf("(R) %s\n", lista->ultimo->pokemon.name);
        free(lista->ultimo);
        lista->ultimo = i;
        lista->ultimo->prox = NULL;
    }
    lista->tam--;
}

void remover(ListaSequencial* lista, int pos) {
    if(lista->primeiro == NULL || pos > lista->tam) {
        printf("Erro");
        exit(1);
    }
    int i;
    Celula* j;
    for(j = lista->primeiro, i = 0; i < pos - 1; i++, j = j->prox);

    Celula* tmp = j->prox;
    j->prox = tmp->prox;
    printf("(R) %s\n", tmp->pokemon.name);
    free(tmp);
    lista->tam--;
}

char * replace(char const * const original, char const * const pattern, char const * const replacement) {
    size_t const replen = strlen(replacement);
    size_t const patlen = strlen(pattern);
    size_t const orilen = strlen(original);

    size_t patcnt = 0;
    const char * oriptr;
    const char * patloc;

    for (oriptr = original; (patloc = strstr(oriptr, pattern)); oriptr = patloc + patlen) {
        patcnt++;
    }

    size_t const retlen = orilen + patcnt * (replen - patlen);
    char * const returned = (char *) malloc(sizeof(char) * (retlen + 1));

    if (returned != NULL) {
        char * retptr = returned;
        for (oriptr = original; (patloc = strstr(oriptr, pattern)); oriptr = patloc + patlen) {
            size_t const skplen = patloc - oriptr;
            strncpy(retptr, oriptr, skplen);
            retptr += skplen;
            strncpy(retptr, replacement, replen);
            retptr += replen;
        }
        strcpy(retptr, oriptr);
    }
    return returned;
}

void limparBuffer() {
    int c;
    while ((c = getchar()) != '\n' && c != EOF);
}

void imprimir(Pokemon pokemon) {
    printf("[#%s -> %s: %s - ", pokemon.id, pokemon.name, pokemon.description);
    if(strcmp(pokemon.types[1], " ") == 0) {
        printf("['%s'] - ", pokemon.types[0]);
    } else {
        printf("['%s', '%s'] - ", pokemon.types[0], pokemon.types[1]);
    }
    printf("%s", pokemon.abilities);
    printf(" - ");
    printf("%skg - %sm - %s%% - %s - %s gen] - %s\n", pokemon.weight,
           pokemon.height, pokemon.captureRate,
           pokemon.isLegendary ? "true" : "false",
           pokemon.generation, pokemon.captureDate);
}

void ler() {
    while (numIds < 801) {
        scanf("%s", ids[numIds]);
        if (strcmp(ids[numIds], "FIM") == 0) {
            return;
        }
        numIds++;
    }
}

Pokemon parsePokemon(char *linha) {
    Pokemon pokemon;
    int contador = 1;
    char type1[30];
    char type2[30] = "";
    char *token = strtok(linha, ",");

    while (token) {
        switch (contador) {
            case 1:
                strcpy(pokemon.id, token);
                break;
            case 2:
                strcpy(pokemon.generation, token);
                break;
            case 3:
                strcpy(pokemon.name, token);
                break;
            case 4:
                strcpy(pokemon.description, token);
                break;
            case 5:
                strcpy(type1, token);
                break;
            case 6:
                strcpy(type2, token);
                break;
            case 7: {
                char *formataHabilidades;
                formataHabilidades = replace(token, ";", ", ");
                strcpy(pokemon.abilities, formataHabilidades);
                break;
            }
            case 8:
                if(strcmp(token, " ") == 0) {
                    strcpy(pokemon.weight, "0.0");
                }else {
                    strcpy(pokemon.weight, token);
                }
                break;
            case 9:
                if(strcmp(token, " ") == 0) {
                    strcpy(pokemon.height, "0.0");
                }else {
                    strcpy(pokemon.height, token);
                }
                break;
            case 10:
                strcpy(pokemon.captureRate, token);
                break;
            case 11:
                pokemon.isLegendary = strcmp(token, "0") != 0;
                break;
            case 12:
                strcpy(pokemon.captureDate, token);
                break;
            default:
                break;
        }
        token = strtok(NULL, ",");
        contador++;
    }

    strcpy(pokemon.types[0], type1);
    if (strlen(type2) > 0) {
        strcpy(pokemon.types[1], type2);
    } else {
        strcpy(pokemon.types[1], "");
    }

    return pokemon;
}

int main() {
    FILE *file = fopen("/tmp/pokemon.csv", "r");

    if (!file) {
        printf("Não foi possível abrir o arquivo.");
        return 1;
    }

    char linha[400];
    ListaSequencial lista;
    Pokemon listaPokemon[803];
    start(&lista);

    ler();

    int cont = 0;
    while (fgets(linha, sizeof(linha), file) != NULL) {
        linha[strcspn(linha, "\n")] = '\0';
        char compareString[400];
        char *auxstring;
        auxstring = replace(linha, ", ", ";");
        auxstring = replace(auxstring, ",,,", ", , ,");
        auxstring = replace(auxstring, ",,", ", ,");
        auxstring = replace(auxstring, "\"", "");

        strcpy(compareString, linha);
        listaPokemon[cont] = parsePokemon(auxstring);
        cont++;
    }

    for (int i = 0; i < numIds; i++) {
        for(int j = 0; j < cont; j++) {
            if(strcmp(ids[i], listaPokemon[j].id) == 0) {
                inserirFim(&lista, listaPokemon[j]);
            }
        }
    }
    
    
    int n;
    scanf("%d", &n);
    limparBuffer();

    for (int i = 0; i < n; i++) {
        char comando[20];
        fgets(comando, sizeof(comando), stdin);
        comando[strcspn(comando, "\n")] = '\0';

        char* token = strtok(comando, " ");
        char* id;
        int pos;

        if(strcmp(token, "II") == 0) {
            id = strtok(NULL, " ");
            for (int j = 0; j < sizeof(listaPokemon); j++) {
                if (strcmp(listaPokemon[j].id, id) == 0) {
                    inserirInicio(&lista, listaPokemon[j]);
                    break;
                }
            }           
        } else if(strcmp(token, "IF") == 0) {
            id = strtok(NULL, " ");
            for (int j = 0; j < sizeof(listaPokemon); j++) {
                if (strcmp(listaPokemon[j].id, id) == 0) {
                    inserirFim(&lista, listaPokemon[j]);
                    break;
                }
            }
        } else if(strcmp(token, "I*") == 0) {
            pos = atoi(strtok(NULL, " "));
            id = strtok(NULL, " ");
            for (int j = 0; j < sizeof(listaPokemon); j++) {
                if (strcmp(listaPokemon[j].id, id) == 0) {
                    inserir(&lista, listaPokemon[j], pos);
                    break;
                }
            }
        } else if(strcmp(token, "RI") == 0) {
            removerInicio(&lista);
        } else if(strcmp(token, "RF") == 0) {
            removerFim(&lista);
        } else if(strcmp(token, "R*") == 0) {
            pos = atoi(strtok(NULL, " "));
            remover(&lista, pos);
        }
    }

    fclose(file);

    Celula* j;
    int posicao;
    for (j = lista.primeiro, posicao = 0; j != NULL; j = j->prox, posicao++) {
        printf("[%d] ", posicao);
        imprimir(j->pokemon);
    }

    return 0;
}